Explorer/Assets/Scripts/MVC/MVCManager.cs

```diff
@@ -0,0 +1,170 @@
+        {
+            // Find the controller
+            IController controller = controllers[typeof(IController<TView, TInputData>)];
+            CancellationToken ct = destructionCancellationTokenSource.Token;
+
+            switch (controller.SortLayers)
+            {
+                case CanvasOrdering.SortingLayer.Popup:
+                    await ShowPopup(command, controller, ct);
+                    break;
+                case CanvasOrdering.SortingLayer.Fullscreen:
+                    await ShowFullScreen(command, controller, ct);
+                    break;
+                case CanvasOrdering.SortingLayer.Persistent:
+                    await ShowPersistent(command, controller, ct);
+                    break;
+                case CanvasOrdering.SortingLayer.Top:
+                    await ShowTop(command, controller, ct);
+                    break;
+            }
+        }
+
+        private async UniTask ShowTop<TView, TInputData>(ShowCommand<TView, TInputData> command, IController controller, CancellationToken ct)
+            where TView : IView
+        {
+            // Push new fullscreen controller
+            TopPushInfo topPushInfo = windowsStackManager.PushTop(controller);
+
+            // Hide all popups in the stack and clear it
+            if(topPushInfo.PopupControllers != null)
+            {
+                foreach (IController popupController in topPushInfo.PopupControllers)
+                    popupController.HideView(ct).Forget();
+
+                topPushInfo.PopupControllers.Clear();
+            }
+
+            // Hide fullscreen UI if any
+            if (topPushInfo.FullscreenController != null)
+            {
+                topPushInfo.FullscreenController.HideView(ct).Forget();
+                windowsStackManager.PopFullscreen(topPushInfo.FullscreenController);
+            }
+
+            // Hide the popup closer
+            popupCloser.Hide(ct).Forget();
+
+            await UniTask.WhenAll(command.Execute(controller, topPushInfo.ControllerOrdering, ct));
+
+            await controller.HideView(ct);
+        }
+
+        private async UniTask ShowPersistent<TView, TInputData>(ShowCommand<TView, TInputData> command, IController controller, CancellationToken ct)
+            where TView : IView
+        {
+            // Push new fullscreen controller
+            PersistentPushInfo persistentPushInfo = windowsStackManager.PushPersistent(controller);
+
+            await UniTask.WhenAll(command.Execute(controller, persistentPushInfo.ControllerOrdering, ct));
+        }
+
+        private async UniTask ShowFullScreen<TView, TInputData>(ShowCommand<TView, TInputData> command, IController controller, CancellationToken ct)
+            where TView : IView
+        {
+            // Push new fullscreen controller
+            FullscreenPushInfo fullscreenPushInfo = windowsStackManager.PushFullscreen(controller);

```

Comment:
What happens with the previous fullscreen?

Reply:
By design there is only 1 fullscreen at a time